build.c

```diff
@@ -290,9 +306,31 @@ jobstart(struct job *j, struct edge *e)
 		description = buildopts.verbose ? NULL : edgevar(e, "description", true);
 		if (!description || description->n == 0)
 			description = j->cmd;
-		formatstatus(status, sizeof(status));
+		if (!dumb)
+			putchar('\r');
+		len = formatstatus(status, sizeof(status));
```

Comment:
The return value here is actually the length needed for the full status (similar to how `snprintf`) works. In the highly unlikely case that it is longer than 256 bytes (`len >= sizeof(status)`), you should set `len = sizeof(status) - 1`.